signalfx/resource_signalfx_alert_muting_rule.go

```diff
@@ -0,0 +1,256 @@
+				Property:      "sf_detectorId",
+				PropertyValue: d.(string),
+				NOT:           false,
+			})
+		}
+	}
+
+	cuamrr := &alertmuting.CreateUpdateAlertMutingRuleRequest{
+		Description: d.Get("description").(string),
+		Filters:     filterList,
+		StartTime:   int64(d.Get("start_time").(int) * 1000),
+		StopTime:    int64(d.Get("stop_time").(int) * 1000),
+	}
+
+	return cuamrr, nil
+}
+
+func alertMutingRuleCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*signalfxConfig)
+	payload, err := getPayloadAlertMutingRule(d)
+	if err != nil {
+		return fmt.Errorf("Failed creating json payload: %s", err.Error())
+	}
+
+	debugOutput, _ := json.Marshal(payload)
+	log.Printf("[DEBUG] SignalFx: Create Alert Muting Rule Payload: %s", string(debugOutput))
+
+	amr, err := config.Client.CreateAlertMutingRule(payload)
+	if err != nil {
+		return err
+	}
+	d.SetId(amr.Id)
+
+	return alertMutingRuleAPIToTF(d, amr)
+}
+
+func alertMutingRuleExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	config := meta.(*signalfxConfig)
+	_, err := config.Client.GetAlertMutingRule(d.Id())
+	if err != nil {
+		if strings.Contains(err.Error(), "404") {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+func alertMutingRuleRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*signalfxConfig)
+	amr, err := config.Client.GetAlertMutingRule(d.Id())
+	if err != nil {
+		if strings.Contains(err.Error(), "404") {
+			d.SetId("")
+		}
+		return err
+	}
+
+	return alertMutingRuleAPIToTF(d, amr)
+}
+
+func alertMutingRuleAPIToTF(d *schema.ResourceData, amr *alertmuting.AlertMutingRule) error {
+	debugOutput, _ := json.Marshal(amr)
+	log.Printf("[DEBUG] SignalFx: Got Alert Muting Rule to enState: %s", string(debugOutput))
+
+	if err := d.Set("description", amr.Description); err != nil {
+		return err
+	}
+
+	if amr.Filters != nil && len(amr.Filters) > 0 {
+		var filters []map[string]interface{}
+		var detectors []string
+		for _, f := range amr.Filters {
+
+			switch f.Property {
+			// The API does not differentiate, but we do to make things
+			// easier for the user, so separate detectors out into their
+			// own propery.
+			case "sf_detectorId":
+				detectors = append(detectors, f.PropertyValue)
+			default:
+				filters = append(filters, map[string]interface{}{
+					"property":       f.Property,
+					"property_value": f.PropertyValue,
+					"negated":        f.NOT,
+				})
+			}
+		}
+		if err := d.Set("filter", filters); err != nil {
+			return err
+		}
+		if err := d.Set("detectors", detectors); err != nil {
+			return err
+		}
+		// The API changes `startTime` to be >= the current
+		// timestamp at the time of the API call. This means
+		// it will pretty much never agree with what the user specified.
+		// To accomodate this we will store the "effective" start time
+		// as a computed attribute, then…
+		if err := d.Set("effective_start_time", amr.StartTime); err != nil {
```

Comment:
You divided `amr.StopTime` by 1000 (I'm guessing converting milliseconds to seconds), but don't here.  Below on line 223 you seem to use effective start time as a second value though, not milliseconds.

Does this need a `/ 1000` here too?

Reply:
Nope. I left a comment about that in the Schema, but the gist is that this field is not user-facing and therefore it's not necessary to do any seconds conversion on it. We only do the seconds conversion on other fields because humans typically thing of Unix time as seconds. No need to do it on this one. (Hopefully the comment helps that for future readers?)